demo/enumApi.ts

```diff
@@ -0,0 +1,590 @@
+      body: user,
+    })
+  );
+}
+/**
+ * Delete user
+ */
+export function deleteUser(username: string, opts?: Oazapfts.RequestOpts) {
+  return oazapfts.fetchText(`/user/${encodeURIComponent(username)}`, {
+    ...opts,
+    method: "DELETE",
+  });
+}
+export function customizePet(
+  {
+    furColor,
+    color,
+    xColorOptions,
+  }: {
+    furColor?: string;
+    color?: string;
+    xColorOptions?: string;
+  } = {},
+  opts?: Oazapfts.RequestOpts
+) {
+  return oazapfts.fetchText(
+    `/pet/customize${QS.query(
+      QS.explode({
+        "fur.color": furColor,
+        color,
+      })
+    )}`,
+    {
+      ...opts,
+      method: "POST",
+      headers: {
+        ...(opts && opts.headers),
+        "x-color-options": xColorOptions,
+      },
+    }
+  );
+}
+export function getIssue31ByFoo(
+  foo: string,
+  {
+    bar,
+    baz,
+    boo,
+  }: {
+    bar?: Schema;
+    baz?: number;
+    boo?: Schema2;
+  } = {},
+  opts?: Oazapfts.RequestOpts
+) {
+  return oazapfts.fetchText(
+    `/issue31/${encodeURIComponent(foo)}${QS.query(
+      QS.explode({
+        bar,
+        baz,
+        boo,
+      })
+    )}`,
+    {
+      ...opts,
+    }
+  );
+}
+export function getObjectParameters(
+  {
+    defaultArray,
+    explodedFormArray,
+    commaArray,
+    defaultSpaceDelimited,
+    explodedSpaceDelimited,
+    spaceDelimited,
+    defaultPipeDelimited,
+    explodedPipeDelimited,
+    pipeDelimited,
+    defaultObject,
+    explodedFormObject,
+    commaObject,
+    deepObject,
+  }: {
+    defaultArray?: Option;
+    explodedFormArray?: Option;
+    commaArray?: Option;
+    defaultSpaceDelimited?: Option;
+    explodedSpaceDelimited?: Option;
+    spaceDelimited?: Option;
+    defaultPipeDelimited?: Option;
+    explodedPipeDelimited?: Option;
+    pipeDelimited?: Option;
+    defaultObject?: Tag;
+    explodedFormObject?: Tag;
+    commaObject?: Tag;
+    deepObject?: Tag;
+  } = {},
+  opts?: Oazapfts.RequestOpts
+) {
+  return oazapfts.fetchText(
+    `/object-parameters${QS.query(
+      QS.explode({
+        defaultArray,
+        explodedFormArray,
+        defaultSpaceDelimited,
+        explodedSpaceDelimited,
+        defaultPipeDelimited,
+        explodedPipeDelimited,
+        defaultObject,
+        explodedFormObject,
+      }),
+      QS.form({
+        commaArray,
+        commaObject,
+      }),
+      QS.space({
+        spaceDelimited,
+      }),
+      QS.pipe({
+        pipeDelimited,
+      }),
+      QS.deep({
+        deepObject,
+      })
+    )}`,
+    {
+      ...opts,
+    }
+  );
+}
+/**
+ * uploads an image in png format
+ */
+export function uploadPng(body?: Blob, opts?: Oazapfts.RequestOpts) {
+  return oazapfts.fetchJson<{
+    status: 200;
+    data: ApiResponse;
+  }>("/uploadPng", {
+    ...opts,
+    method: "POST",
+    body,
+  });
+}
+export enum Status {
+  Available = "Available",
+  Pending = "Pending",
+  Sold = "Sold",
+}
+export enum Animal {
+  true = "true",
```

Comment:
A better question would be how a boolean enum is useful on the OpenAPI spec. Could not find anything about that.

The exported enum seems to work with the boolean as the name of the property. But I agree it does not make much sense to use a boolean as an enum key.
<img width="1204" alt="image" src="https://user-images.githubusercontent.com/1686408/196464632-07fcfb32-0747-42a0-a7c6-f4d0ba427081.png">

I changed the code so that it would use the previous union type in the case of a boolean enum. But would be nice to know what OpenAPI enum boolean actually should mean.

Reply:
I agree that this is probably a unicorn-case that should never happen in real-live. Thanks for taking care of it 🦄 👍